aten/src/ATen/native/cuda/DistributionTemplates.h

```diff
@@ -51,10 +51,31 @@ std::tuple<uint64_t, dim3, dim3> calc_execution_policy(int64_t total_elements) {
   const uint32_t unroll = curand4_engine_calls;
   dim3 dim_block(block_size);
   dim3 grid((numel + block_size - 1) / block_size);
-  uint32_t blocks_per_sm = at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor / block_size;
-  grid.x = std::min(
-      static_cast<uint32_t>(at::cuda::getCurrentDeviceProperties()->multiProcessorCount) * blocks_per_sm,
-      grid.x);
+
+
+  // We changed the offset calculations to be independent of CUDA devices
+  // properties. Earlier the impl was
+  //
+  // max_threads_per_sm = static_cast<uint32_t>(at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor);
+  // number_of_sm = static_cast<uint32_t>(at::cuda::getCurrentDeviceProperties()->multiProcessorCount);
+  //
+  // However, having the offset dependent on cuda device properties makes it
+  // harder/hacky to support functionalization of RNG ops. So we have chosen
+  // lower bounds for the two device properties by looking at the recent GPUs.
+  // Lower bounds ensure that we move the offsets more aggressively, ensuring
+  // that random numbers are not repeated.
+
+
+  // Newer GPUs have 2048 threads per SM, but older ones have 1536. We chose
```

Comment:
it's not newer/older, there's a weird thing going with oscillating threads. 
You could also mention that philox sequence length is `2**64` so we are unlikely to run out, and can advance liberally.